### PR TITLE
fix up core graphics full-screen animations

### PR DESCRIPTION
**Pull request by osheroff**
This improves a "flickering" seen when entering and exiting full-screen, which is due to the layer code incorrectly drawing the non-fullscreen layer (often with the vim command `set invfu`) as an intermediate state.  Here we just tell the layer code to draw an empty screen until it receives the "redraw-the-screen" batch of commands from vim.

This solution isn't 100%, as it doesn't seem possible to control just when cocoa actually does work; sometimes the resize of the window doesn't take affect until some milliseconds after we start fading up into fullscreen, which means you catch a dim view of the old title bar whilst fading up.  Conversely, sometimes when exiting fullscreen the resize event hasn't taken place yet and you end 
fading out and then still painting the fullscreen window.

In short; I think this improves things, but not all the way.

### CODE REVIEW

**Comment by splhack** [comment]
could you test this with `set transparency=10`?

**Comment by osheroff** [comment]
meh.  well, this PR doesn't make the situation any worse, but as it turns out the `CGLayer` code broke transparency, generally.

**Comment by splhack** [comment]
How about moving shouldBlankUntilRedraw code block in cgLayerEnabled until supporting transparency in CGLayer rendering?

    if (cgLayerEnabled) {
        if (shouldBlankUntilRedraw) {
            ...

**Comment by osheroff** [comment]
sure, I can do that.  I don't honestly think it matters one way or another but it's certainly a way to not incur any unwanted side-effects.

The deeper I go down these paths, the less sure of the `CGLayer` approach I get.  You're right that there's a little lag incurred -- copying the layer over into the window is by far the most expensive part of `drawRect`.  I wonder if maybe we wouldn't be better off with a "traditional" render path here: `performBatchDrawWithData` would just update a data-structure and call `setNeedsDisplay`, which would draw to the screen from the text-buffer.  Whether that can be made faster than `NSTextView`, I dunno.  What do you think?

**Comment by splhack** [comment]
well, if I had known the way, the issue would have been already fixed :)

**Comment by osheroff** [comment]
i'm kinda convinced that there should be a better way than this.  It doesn't really make it all the way better so I'm gonna close this PR and try a different tack.
